internal/wire/records_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wire
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+// loadRecords gets recursively all .bin files from the recordsPath directory,
+// parses their content to wire Msgs and returns them as an array of testCase
+// structs with headerB and bodyB fields set.
+// If no records are found, it returns an empty slice and no error.
+func loadRecords(recordsPath string) ([]testCase, error) {
+	// Load recursively every file path with ".bin" extension from recordsPath directory
+	var recordFiles []string
+
+	err := filepath.WalkDir(recordsPath, func(path string, entry fs.DirEntry, err error) error {
+		if err != nil {
```

Comment:
Do we need that error check here?

Reply:
```go
func WalkDir(root string, fn fs.WalkDirFunc) error {
	info, err := os.Lstat(root)
	if err != nil {
		err = fn(root, nil, err)
	} else {
		err = walkDir(root, &statDirEntry{info}, fn)
	}
	if err == SkipDir {
		return nil
	}
	return err
}

```
It pass the error to the function if Lstat failed. So, yeah I think that's important to check for it

Reply:
Yes, that's right but I can't see `Lstat` call 😃

Reply:
`Walkdir` does it in the first line. If it fails, it pass an error to the function provided as an argument

Reply:
`err` is passed to the walk function for advanced error handling, but `if err != nil { return err }` is the most common implementation. In fact, I don't think I ever had any other implementation :D

Reply:
Alternatively, we might want to move `os.IsNotExist` check there

Reply:
Sounds good